athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchFieldResolver.java

```diff
@@ -0,0 +1,275 @@
+/*-
+ * #%L
+ * athena-elasticsearch
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch;
+
+import com.amazonaws.athena.connector.lambda.data.FieldResolver;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Used to resolve Elasticsearch complex structures to Apache Arrow Types.
+ * @see com.amazonaws.athena.connector.lambda.data.FieldResolver
+ */
+public class ElasticsearchFieldResolver
+        implements FieldResolver
+{
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchTypeUtils.class);
+
+    protected ElasticsearchFieldResolver() {}
+
+    /**
+     * Return the field value from a complex structure or list.
+     * @param field is the field that we would like to extract from the provided value.
+     * @param originalValue is the original value object.
+     * @return the field's value as an ArrayList for a LIST field type, a HashMap for a STRUCT field type, or the actual
+     * value if neither of the above.
+     * @throws RuntimeException if the fieldName does not exist in originalValue, if the fieldType is a STRUCT and
+     * the fieldValue is not instance of HashMap, or if the fieldType is neither a LIST or a STRUCT but the fieldValue
+     * is instance of HashMap (STRUCT).
+     */
+    @Override
+    public Object getFieldValue(Field field, Object originalValue)
+            throws RuntimeException
+    {
+        Types.MinorType fieldType = Types.getMinorTypeForArrowType(field.getType());
+        String fieldName = field.getName();
+        Object fieldValue;
+
+        if (originalValue instanceof HashMap && ((Map) originalValue).containsKey(fieldName)) {
```

Comment:
`instanceof Map` should be sufficient for all occurrences in this method

Reply:
and should throw an IllegalArgumentException if not an instance of Map

Reply:
Fixed